feat: add C implementation for `stats/base/dists/levy/cdf`

AadishJ opened the pull request:
Resolves #3722.

## Description

> What is the purpose of this pull request?

This pull request:

- Adds a C implementation of the levy cdf.
- Includes benchmarks for the C implementation.
- Provides examples demonstrating the usage of the levy cdf in C.


## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #3722

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

- [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

Commit:
feat(levy-cdf): adding all functionality required for levy cdf

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/levy/cdf/README.md:
````diff
@@ -139,6 +139,104 @@ for ( i = 0; i < 100; i++ ) {
+### Examples
+
+```c
+#include "stdlib/stats/base/dists/arcsine/cdf.h"
````
Copy-paste error with examole being for arcsine dsitribution.

Planeshifter reviewed (changes requested):
Thanks for your PR!

Looking good overall, but please fix the example in README.

Commit:
feat(levy-cdf): fixing readme

AadishJ commented:
Made the changes please check

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/levy/cdf/benchmark/benchmark.js:
```diff
@@ -21,28 +21,37 @@
+	mu = new Float64Array( len );
+	c = new Float64Array( len );
+	for ( i = 0; i < len; i++ ) {
+		x[ i ] = ( randu() * 20.0 ) - 10.0;
```
@AadishJ We can directly create arrays of random numbers from the desired distribution via `uniform` here for better clarity and to avoid the for-loop.

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/levy/cdf/benchmark/benchmark.native.js:
```diff
@@ -0,0 +1,73 @@
+	mu = new Float64Array( len );
+	c = new Float64Array( len );
+	for ( i = 0; i < len; i++ ) {
+		x[ i ] = ( randu() * 20.0 ) - 10.0;
```
@AadishJ We can directly create arrays of random numbers from the desired distribution via `uniform` here for better clarity and to avoid the for-loop.

AadishJ committed:
feat(levy-cdf): replaced for loop with uniform function to create arrays

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: passed
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

AadishJ commented:
Hey I have made the changes please check

Planeshifter committed:
Merge branch 'develop' into develop9

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter committed:
style: add newline at end of file

Planeshifter committed:
bench: remove Float64Array usage as not needed

Planeshifter commented:
/stdlib update-copyright-years

stdlib-bot committed:
chore: update copyright years

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/levy/cdf/src/main.c:
```diff
@@ -0,0 +1,46 @@
+#include "stdlib/math/base/assert/is_nan.h"
+#include "stdlib/math/base/special/erfc.h"
+#include "stdlib/math/base/special/sqrt.h"
+#include "stdlib/stats/base/dists/levy/cdf.h"
```
```suggestion
#include "stdlib/stats/base/dists/levy/cdf.h"
#include "stdlib/math/base/assert/is_nan.h"
#include "stdlib/math/base/special/erfc.h"
#include "stdlib/math/base/special/sqrt.h"
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/levy/cdf/src/main.c:
```diff
@@ -0,0 +1,46 @@
+* @return     evaluated CDF
+*
+* @example
+* double y = stdlib_base_levy_cdf( 9.0, 0.0, 1.0 );
+* // returns ~0.999
+*/
+double stdlib_base_levy_cdf( const double x, const double mu, const double c ) {
```
```suggestion
* double y = stdlib_base_dists_levy_cdf( 9.0, 0.0, 1.0 );
* // returns ~0.999
*/
double stdlib_base_dists_levy_cdf( const double x, const double mu, const double c ) {
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/levy/cdf/include/stdlib/stats/base/dists/levy/cdf.h:
```diff
@@ -0,0 +1,38 @@
+/**
+* Evaluates the cumulative distribution function (CDF) for a Lévy distribution with location parameter `mu` and scale parameter `c` at a value `x`.
+*/
+double stdlib_base_levy_cdf( const double x, const double mu, const double c );
```
```suggestion
double stdlib_base_dists_levy_cdf( const double x, const double mu, const double c );
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/levy/cdf/README.md:
```diff
@@ -139,6 +139,104 @@ for ( i = 0; i < 100; i++ ) {
+        x = random_uniform( -10.0, 10.0 );
+        mu = random_uniform( -20.0, 0.0 );
+        c = random_uniform( 0.1, 10.0 );
+        y = stdlib_base_levy_cdf( x, mu, c );
```
```suggestion
        y = stdlib_base_dists_levy_cdf( x, mu, c );
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/levy/cdf/examples/c/example.c:
```diff
@@ -0,0 +1,42 @@
+		x = random_uniform( -10.0, 10.0 );
+		mu = random_uniform( -20.0, 0.0 );
+		c = random_uniform( 0.1, 10.0 );
+		y = stdlib_base_levy_cdf( x, mu, c );
```
```suggestion
		y = stdlib_base_dists_levy_cdf( x, mu, c );
```

Planeshifter committed:
fix: use correct function name

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter committed:
chore: reorder includes

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/levy/cdf/benchmark/c/benchmark.c:
```diff
@@ -0,0 +1,142 @@
+* limitations under the License.
+*/
+
+#include <sys/time.h>
+#include "stdlib/stats/base/dists/levy/cdf.h"
+#include <math.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
```
```suggestion
#include "stdlib/stats/base/dists/levy/cdf.h"
#include <sys/time.h>
#include <math.h>
#include <stdio.h>
#include <stdlib.h>
#include <time.h>
```

Planeshifter committed:
chore: reorder includes

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter reviewed (dismissed):
Thanks, @AadishJ!

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/levy/cdf/src/addon.c:
```diff
@@ -0,0 +1,23 @@
+#include "stdlib/stats/base/dists/levy/cdf.h"
+
+// cppcheck-suppress shadowFunction
+STDLIB_MATH_BASE_NAPI_MODULE_DDD_D( stdlib_base_levy_cdf )
```
```suggestion
STDLIB_MATH_BASE_NAPI_MODULE_DDD_D( stdlib_base_dists_levy_cdf )
```

Planeshifter committed:
fix: use correct function name

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/levy/cdf/benchmark/c/benchmark.c:
```diff
@@ -0,0 +1,142 @@
+
+	t = tic();
+	for ( i = 0; i < ITERATIONS; i++ ) {
+		y = stdlib_base_levy_cdf( x[ i % 100 ], mu[ i % 100 ], c[ i % 100 ] );
```
```suggestion
		y = stdlib_base_dists_levy_cdf( x[ i % 100 ], mu[ i % 100 ], c[ i % 100 ] );
```

Planeshifter committed:
fix: use correct function name

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>
